python-modules/cis_crypto/cis_crypto/operation.py

```diff
@@ -97,42 +99,37 @@ def _get_public_key(self):
 
     def _reduce_keys(self, well_known_response):
         access_file_keys = well_known_response['access_file']['jwks_keys']
-        publisher_keys = well_known_response['publishers_supported']
+        publishers_supported = well_known_response['publishers_supported']
 
         keys = []
-        for i in range(len(access_file_keys)):
-            keys.append(
-                {
-                    'access_file_key_{}'.format(i): access_file_keys.get(str(i))
-                }
-            )
 
-        for k in publisher_keys:
-            keys.append(
-                {
-                    k: publisher_keys[k]
-                }
-            )
+        if 'access-file-key' in self.config('public_key_name', namespace='cis', default='access-file-key'):
+            return access_file_keys
+        else:
+            # If not an access key verification this will attempt to verify against any listed publisher.
+            keys = publishers_supported[publisher]['jwks_keys']
+            for key in range(len(keys)):
+                keys.append(
+                    key
+                )
         return keys
 
     def jws(self):
         """Assumes you loaded a payload.  Return the same jws or raise a custom exception."""
         key_material = self._get_public_key()
-        if isinstance(key_material, dict):
-            sig = jws.verify(self.jws_signature, key_material, algorithms='RS256', verify=True)
-            return sig
-        elif isinstance(key_material, list):
+
+        if isinstance(key_material, list):
             logger.debug('Multiple keys returned.  Attempting match.')
             for key in key_material:
-                for k_id in key_material[key_material.index(key)]:
-                    logger.info('Attempting to match against: {}'.format(k_id))
-                    key_content = key_material[key_material.index(key)][k_id]
-                    try:
-                        sig = jws.verify(self.jws_signature, key_content, algorithms='RS256', verify=True)
-                        logger.info('Matched a verified signature for: {}'.format(k_id))
-                        return sig
-                    except JWSError as e:
-                        logger.error(e)
+                key.pop('x5t', None)
```

Comment:
note i dont think we care about `x5t`. its fine to leave it in there, but i dropped it from the schema because its not very useful
x5t being the thumbprint (a sha1 hash of the cert)
x5c being the actual cert (which you can do a .. sha256 hash of.. or sha1)